scenarios/getting_started/rust/getting_started/src/main.rs

```diff
@@ -0,0 +1,161 @@
+use conn::connection_settings;
+use dotenv::dotenv;
+use paho_mqtt as mqtt;
+use paho_mqtt::SslOptionsBuilder;
+use paho_mqtt::SslVersion;
+use paho_mqtt::MQTT_VERSION_5;
+use std::time::Duration;
+
+fn main() {
+    // Initialize the logger from the environment
+    env_logger::init();
+
+    dotenv().ok();
+    let conn_settings = connection_settings::get_connection_settings(Some(".env")).unwrap();
+    let mqtt_clean_session = conn_settings.mqtt_clean_session;
+    if !mqtt_clean_session {
+        println!("This sample does not support connecting with existing sessions");
+        std::process::exit(1);
+    }
+    let port = conn_settings.mqtt_tcp_port;
+    let hostname = conn_settings.mqtt_host_name;
+    let keepalive = conn_settings.mqtt_keep_alive_in_seconds;
+    let client_id = conn_settings.mqtt_client_id;
+    let mut protocol = "tcp";
+
+    // Set up connection builder
+    let mut conn_opts_builder = mqtt::ConnectOptionsBuilder::with_mqtt_version(MQTT_VERSION_5);
+    conn_opts_builder
+        .keep_alive_interval(Duration::from_secs(keepalive.into()))
+        .clean_start(mqtt_clean_session);
+
+    if let Some(username) = &conn_settings.mqtt_username {
+        conn_opts_builder.user_name(username);
+    }
+
+    if let Some(password) = &conn_settings.mqtt_password {
+        conn_opts_builder.password(password);
+    }
+
+    if conn_settings.mqtt_use_tls {
```

Comment:
we should move most of this methods to `mqttclients`, otherwise we will need to duplicate the same code in all the other samples.

Reply:
i followed how we did connection settings for python and this file is absolutely close to what python has

Reply:
I have created a separate task for this to tackle later on. https://dev.azure.com/msazure/One/_workitems/edit/26929393